### PR TITLE
CHANGE (DevOps) @W-16125570@ Feature Branch PR titles can omit scope.

### PR DESCRIPTION
**Pull request by jfeingold35**

### CODE REVIEW

File: github-actions/verify-pr-title/test/verifyFeaturePrTitle.test.ts [hunk 1]
```diff
@@ -103,42 +102,42 @@ describe('#verifyFeaturePrTitle', () => {
...
 	describe('Scope portion', () => {
 		function createTitle(scope: string): string {
 			return `NEW ${scope} @W-1234@ Filler text`;
 		}
 
-		it('Scope portion cannot be absent', () => {
+		it('Scope portion may absent', () => {
 			const title = createTitle('');
-			expect(verifyFeaturePrTitle(title)).to.equal(false, 'Should be rejected');
+			expect(verifyFeaturePrTitle(title)).toEqual(true);
```
Comment (jfeingold35) [inline github-actions/verify-pr-title/test/verifyFeaturePrTitle.test.ts:115]: This is the only test that changed non-cosmetically. All other tests in all other files were simply converted from `chai`'s `expect` implementation to `jest`'s.

File: github-actions/verify-pr-title/src/verifyFeaturePrTitle.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ const SCOPE_PORTION = "\\([^()]+\\)";
  * i.e., a PR aimed at {@code dev} or a {@code release-x.y.z} branch, not
  * coming from {@code main}.
  */
-const FEATURE_PR_REGEX = new RegExp(`^${PR_TYPE_PORTION}${SEPARATOR}${SCOPE_PORTION}${SEPARATOR}${WORK_ITEM_PORTION}${SEPARATOR}[^\\s]+.*`, "i");
+const FEATURE_PR_REGEX = new RegExp(`^${PR_TYPE_PORTION}(${SEPARATOR}${SCOPE_PORTION})?${SEPARATOR}${WORK_ITEM_PORTION}${SEPARATOR}[^\\s]+.*`, "i");
```
Comment (jfeingold35) [inline github-actions/verify-pr-title/src/verifyFeaturePrTitle.ts:20]: Took the "wrap the separator and scope part in a capture group" idea from Stephen's implementation of this functionality in the Core repo.

File: github-actions/verify-pr-title/src/index.ts [hunk 3]
```diff
@@ -53,7 +53,7 @@ function run(): void {
 				console.log(`PR title '${title}' accepted for ${baseBranch} branch`);
 			} else {
 				core.setFailed(
-					`PR title '${title}' does not match the template of "NEW|FIX|CHANGE (__) @W-XXXX@ Summary"`
+					`PR title '${title}' does not match the template of "NEW|FIX|CHANGE (__) @W-XXXX@ Summary" or "NEW|FIX|CHANGE @W-XXXX@ Summary"`
```
Comment (jfeingold35) [inline github-actions/verify-pr-title/src/index.ts:56]: Couldn't think of a way to clearly articulate that the scope part was optional within a single string, so I just added a second part.

File: github-actions/verify-pr-title/src/common.ts [hunk 4]
```diff
@@ -11,5 +11,5 @@ export const WORK_ITEM_PORTION = "@W-\\d{4,9}@"
  * colon, or semicolon characters. All pull request titles allow these
  * characters as separators between portions
  */
-export const SEPARATOR = "[ -.,;:]*";
+export const SEPARATOR = "[\\s\\-\\.\\,\\;\\:]*";
```
Comment (jfeingold35) [inline github-actions/verify-pr-title/src/common.ts:14]: The online regex tester I used when I was playing around with this didn't like the unescaped special characters, so I just went ahead and escaped them.

File: github-actions/verify-pr-title/package.json [hunk 5]
```diff
@@ -8,24 +8,37 @@
 		"build": "tsc",
 		"lint": "eslint src/**/*.ts",
 		"package": "npx ncc build",
-		"test": "mocha \"./test/**/*.test.ts\"",
+		"test": "jest",
```
Comment (jfeingold35) [inline github-actions/verify-pr-title/package.json:11]: For whatever reason, Mocha _really_ didn't like running inside a subfolder where the parent folder was using Jest. So I just went ahead and converted the tests to Jest.

File: github-actions/verify-pr-title/package.json [hunk 6]
```diff
@@ -8,24 +8,37 @@
...
-		"@types/mocha": "^7.0.2",
 		"@types/node": "^20",
 		"@typescript-eslint/eslint-plugin": "^7.2.0",
 		"@typescript-eslint/parser": "^7.2.0",
 		"@vercel/ncc": "^0.38.1",
 		"eslint": "^8.57.0",
-		"mocha": "^10.4.0",
-		"mocha-junit-reporter": "^1.23.3",
-		"typescript": "^5.0.0"
+		"typescript": "^5.5.2"
```
Comment (jfeingold35) [inline github-actions/verify-pr-title/package.json:42]: I needed to add `typescript` so the tests would actually compile.

File: github-actions/verify-pr-title/package.json [hunk 7]
```diff
@@ -8,24 +8,37 @@
...
 		"all": "npm run build && npm run lint && npm test && npm run package"
 	},
 	"dependencies": {
 		"@actions/core": "^1.10.1",
-		"@actions/github": "^6.0.0"
+		"@actions/github": "^6.0.0",
+		"jest": "^29.7.0",
+		"ts-jest": "^29.1.5"
+	},
+	"jest": {
```
Comment (jfeingold35) [inline github-actions/verify-pr-title/package.json:20]: This is all copy-pasted from the outer `package.json`, sans the code coverage stuff.

**Review by stephen-carter-at-sf — APPROVED** [review]

**Pull request merged**